Frontend

doxikus opened the pull request:
Hi,

I have created new branch where are only updates related to template changes, adding some HTML tags, also meta viewport for mobile devices, it is needed because I want to create few custom css skins so user can decide which one they want to use/upload. Also I have updated TD, TR, TH and make them in lowercase.

Hope you will accept this request and merge so I can continue to work on CSS skin next

Best regards

Commit:
Updating template structure, adding viewport meta name for mobile devices, feature ready for responsive layout

Commit:
Adding meta viewport, feature ready nad support for mobile devices, responsive layout, update template structure, adding article html5 tag, easier to style interface, find/replace uppercase to lowercase, matter of elegance and preference code readability :-)

Commit:
conflict resolve

TD-er commented:
I guess adding the HTML tags and meta viewport things is a (very) good idea.
Is changing TD/TR/TH into lower case really needed according to standards?
The reason I ask is that there is also a major pull request ( #850 ) for changing the webserver file into a streaming variant and perhaps merging all those might be quite difficult.
And I want to keep these things a little bit close to v2.0, so I will add these to the v2.0 branch also.
It looks like it won't break anything yet, I guess?

Just to make sure merging between both versions is not made impossible until we release v2.0

I will look into this this evening (or @psy0rz will do it sooner, if his fingers are itching to do so ;) ).
And good idea to allow for a better view on mobile devices. I guess those will be used the most for ESPeasy configuration.

doxikus commented:
Ola, as for lowercase/upper case there is no difference today, browser will render them equally, only in time of xHTML standard lowercase was must, more closer to XML standard. Today I'm to lazy to keep eye on caps lock :-).

s0170071 commented:
Looks like we're faster doing search and replace than a merge with regard to #850.
@TD-er what do you think how long it will be until #850 is merged ?

TD-er commented:
@s0170071 I had no chance yet to test it, but @psy0rz was already having a look at it.
We might want to merge it to a separate branch to test first and then merge into the main branches?
Then more people can test it.

psy0rz commented:
@doxikus please undo the case-conversion and only make changes that are absolutely necessary. so no spelling changes, white space changes or case conversion.

Otherwise we will get an impossible to merge mess with the 2k webserver stuff @TD-er was talking about.

doxikus commented:
Hi, 

@s0170071 well this was the easiest job :-)

@psy0rz sure, I will undo all unnecessary changes and add only one related to template update.

Again sorry to all about rushing in without thinking about others, won't happen again.

TD-er commented:
Well, the current situation is a bit undesired with such big PR's on the verge of being merged.
We still haven't decided yet if the "2k-webserver' PR will be in v2.0 branch.
If it is going to be backported to v2.0 then merging is not really an issue anymore, but if we decide to make that PR be the "merge-breaker", then things will change regarding merges.

Commit:
revert to default state and change only template related lines

Commit:
Merge branch 'frontend' of github.com:doxikus/ESPEasy into frontend

Commit:
final check, revert uppercase

Commit:
Last template changes, lost in undo process :-)

doxikus commented:
Ok, I have reverted back changes, added only template related files, now is it possible to keep this pull request on hold, I have made changes on markup and web interface will "explode" because CSS is not following new template hierarchy. Just keep it on frontend branch for now, I need to play little bit with platformio so I can generate FW for testing local, anyone can build from this branch, I will send CSS so it can be uploaded trough tools/settings/load. Later if we decide to have new default theme, css can be loaded inside this file like now, starting from line 94 on WebServer.ino

uzi18 commented:
Please revert this also:    
`"<h6>Powered by www.letsconTRolit.com</h6>"`

doxikus commented:
@TD-er Well as I understand there is a lot of changes regarding to WebServer.ino file, for me it is not a problem I can wait till more important things is resolved, I can always update part which is related to template and markup, even add all changes again. I want to have code ready so I can continue with css skins. I will for sure continue to work on my version of FW and then commit all changes back here.

Commit:
h6 footer TR

doxikus commented:
@uzi18 done, tnx

uzi18 commented:
Thanks

psy0rz commented:
@doxikus its no problem, thats why we have PR's where we can request changes and discuss things.

We're happy you're willing to help, thanks!
